### PR TITLE
[4.0] Fixing issues from Psalm quality checks

### PR DESCRIPTION
**Pull request by Hackwar**
This is another batch of fixes to docblocks and method calls from the Psalm quality tool. This can not be tested, but needs to be reviewed.

### CODE REVIEW

**Comment by PhilETaylor** [comment]
Careful... you will get dragged down a huge rabbit hole updating the docblocks and the types of each param... literally thousands of them are wrong. I looked to fix them but was overwhelmed! Making joomla type safe is going to be a nightmare in the future.

**Comment by Hackwar** [comment]
Which is why I'm starting small and doing little steps. Not claiming to fix every problem everywhere, but when I run into them, I can fix them. Every improvement helps when later working with an IDE.

**Comment by PhilETaylor** [comment]
Totally agree. phpStorm has a nightmare at the moment  :-(

File: libraries/src/Cache/Controller/CallbackController.php [hunk 1]
```diff
@@ -183,7 +183,7 @@ public function store($data, $id, $group = null, $wrkarounds = true)
 	/**
 	 * Generate a callback cache ID
 	 *
-	 * @param   callback  $callback  Callback to cache
```
Comment (wilsonge) [inline libraries/src/Cache/Controller/CallbackController.php:186]: Whilst obviously callback isn't a thing. This new block isn't true either as you can see from the first line of the function it handles objects and array's that aren't direct callable's. I think until we look closer for now this just needs to be `mixed`

File: libraries/src/Component/Router/RouterViewConfiguration.php [hunk 2]
```diff
@@ -147,7 +147,7 @@ public function setKey($key)
 	 *
 	 * @since   3.5
 	 */
-	public function setParent(RouterViewconfiguration $parent, $parentKey = false)
+	public function setParent(RouterViewconfiguration $parent, $parentKey = null)
```
Comment (wilsonge) [inline libraries/src/Component/Router/RouterViewConfiguration.php:150]: Why not just an empty string and use `empty()`?

**Comment by wilsonge** [comment]
One fix required and this is good to merge

File: libraries/src/Cache/Controller/CallbackController.php (hunk 1 above)
Comment (Hackwar) [reply libraries/src/Cache/Controller/CallbackController.php:186]: fixed

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**